client/lib/wp/sync-handler/index.js

```diff
@@ -187,13 +193,48 @@ export class SyncHandler {
 		return localforage.getItem( key, fn );
 	}
 
+	/**
+	 * Add/Override a record.
+	 *
+	 * @param {String} key - record key identifier
+	 * @param {Object} data - data to store
+	 * @param {Function} [fn] - callback function
+	 * @return {Promise} natuve ES6 promise
+	 */
 	storeRecord( key, data, fn = () => {} ) {
 		debug( 'storing data in %o key\n', key );
-		return localforage.setItem( key, data, fn );
+
+		// add this record to history
+		return cacheInvalidation
```

Comment:
> I feel like this is so closely coupled that we can let the cache also set the record here, rather than doing it in a callback below. (The same also applies to removeRecord)

The method `cache.addItem( key )` adds an item into the `records-list` record. This item has the couple of `key` and `mark` values:

```es6
cache.add( key ) // -> { key, mark: Date.now() } added to `records-list` record
```

If we wanna add this item and at the same call add the new record to local data we will have to pass the `key` and the request/response object (`data`) to same method:


Thus we should change the name of the method to something like 
```es6
// Adds { key, mark: Date.now() } to `records-list` record
// and adds `sync-record-<hex-value>` record to local data.
cache.addItemAndStoreRecord( key, data )
```